docs/sect-finality.md

```diff
@@ -485,3 +485,268 @@ A Catch-up response message contains critical information for the requester node
+
+Therefore, given an MMR tree with n peaks ordered in decreasing order of height, the `MMR-root` of the tree is calculated as follows.
+
+$$
+\texttt{MMR root} = hash (p_1 + hash( p_2 + hash( p_3 +... + hash ( p_n-1 + p_n ))))
+$$
+
+**where**
+- $p_1, p_2,...,p_n$: represents the list of current peaks in decreasing order of heights.
+- $hash$: corresponds to the 256-bit [Keccak](chap-host-api#sect-finality#id-ext_hashing_keccak_256) hash function used to merge the peaks.
+::::
+
+A distinguished feature of this process is that whenever new leaf nodes are added to the tree, the earlier hash computations of peaks are reused, making new leaf nodes less expensive to insert and to prove (i.e., to verify the integrity of leaf data). 
+
+
+###### Definition -def-num- MMR operations {#defn-mmr-operations}
+::::definition
+
+Here are the basic operations we should be able to perform on the MMR:
+
+- Append Leaf Node (appendData):
+    - Signature: append(data: T) -> None
+    - Description: appends a new leaf element with the provided data to the MMR.
+		  		  	  
+- Create MMR root (bagPeaks):
+    - Signature: baggingPeaks(peaksIndexes: List[int]) -> str
+    - Description: creates the single MMR root based on the list of peaks, and returns the hash string of the  MMR root corresponding to the current state of the tree. 
+
+- Verify Node (verifyProof):
+    - Signature: verifyNode(nodeHash: str, requiredProofNodes: List[str], MMRroot: str) -> bool
+    - Description: verifies if the given node hash can be proved based on the list of required proof nodes and the MMR root hash.
+:::
+
+###### Definition -def-num- Payload {#defn-beefy-payload}
+:::definition
+
+**Payload:** is the Merkle root of the MMR generated where the leaf data contains the following fields for each block:
+- _LeafVersion_: a byte indicating the current version number of the Leaf Format. The first 3 bits are for major versions and the last 5 bits are for minor versions. 
+- _BeefyNextAuthoritySetInfo_: It is a tuple consisting of:
+    - ValidatorSetID
+    - Len (u32): length of the validator set
+    - Merkle Root of the list of Next Beefy Authority Set (ECDSA public keys). The exact format depends on the implementation.  
+- Parent Block number and Parent Block Hash.
+- Extra Leaf Data: Currently the Merkle root of the list of (ParaID, ParaHeads)
+:::
+
+
+###### Definition -def-num- Commitment {#defn-beefy-Commitment}
+:::definition
+
+**Signed Commitment:**
+`commitment` is a tuple of `(payload, Block Number, ValidatorSetID)`. A `signed commitment` is a tuple `(commitment, signatures)`, where `signatures` is a list of optional signatures of the validator set on the SCALE encoded `commitment`. Note that the number of signatures in `signatures` may be less than the length of the Validator Set. 
```

Comment:
The Definition is called "Commitment" but what is in fact defined is a "Singed Commitment".  Wouldn't it be better to either divide it into two distinguished definitions, or simply call that "Signed Commitment" and make the definition of commitment an inner preliminary definition of